fontforge/python.c

```diff
@@ -17893,14 +17894,18 @@ return (NULL);
     for ( i=0; i<map->enccount; ++i ) if ( (gid=map->map[i])!=-1 && (sc=sf->glyphs[gid])!=NULL && fv->selected[i] ) {
 	changed = refchanged = false;
 	if ( checkrefs ) {
-	    for ( ref=sc->layers[self->fv->active_layer].refs; ref!=NULL; ref=ref->next ) {
+	    for ( ref=sc->layers[self->fv->active_layer].refs; ref!=NULL; ) {
 		if ( ref->transform[0]*ref->transform[3]<0 ||
 			(ref->transform[0]==0 && ref->transform[1]*ref->transform[2]>0)) {
 		    if ( !refchanged ) {
 			refchanged = true;
 			SCPreserveLayer(sc,self->fv->active_layer,false);
 		    }
+		    next = ref->next;
 		    SCRefToSplines(sc,ref,self->fv->active_layer);
```

Comment:
Does `SCRefToSplines` invalidate `ref`? If so, we need to figure out what's going wrong there.

Reply:
Oh. I see. `SCRemoveDependent` calls `SCRemoveDependent` which calls `RefCharFree`.

Reply:
Or, rather `SCRefToSplines` calls `SCRemoveDependent` which calls `RefCharFree`.